spec/server/listener_metrics_spec.rb

```diff
@@ -170,12 +170,16 @@ module Backend
           end
         end
 
-        context 'running Puma' do
-          it_behaves_like 'listener with metrics', :puma
+        unless ThreeScale::Backend.configuration.redis.async
+          context 'running Puma' do
+            it_behaves_like 'listener with metrics', :puma
+          end
         end
 
-        context 'running Falcon' do
-          it_behaves_like 'listener with metrics', :falcon
+        if ThreeScale::Backend.configuration.redis.async
```

Comment:
if/else maybe instead of two independent conditional blocks?

Reply:
same applies further down

Reply:
Right